### PR TITLE
Use `Intl.Collator` when possible

### LINKED ISSUE #1: Use `Intl.Collator` when available in Node
Per [MDN recommendation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#Performance):

> When comparing large numbers of strings, such as in sorting large arrays, it is better to create an [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Collator) object and use the function provided by its [compare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Collator/compare) property.
## 

Will also need a fallback for older Node versions and the browser.

### PR DESCRIPTION
**Pull request by nuno-morais**
Fixes #1

### CODE REVIEW

**Comment by satazor** [comment]
@sindresorhus tests are failing on node v0.10 and v0.12 because of `xo`. Can we remove them?

File: index.js [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-	if (brokenLocaleCompare) {
+if (brokenLocaleCompare) {
+	compare = function (a, b) {
 		return a > b ? 1 : a < b ? -1 : 0;
-	}
-
-	return a === b ? 0 : a.localeCompare(b);
+	};
+} else if (typeof Intl.Collator === 'function') {
+	var collator = new Intl.Collator({sensitivity: 'variant'});
```
Comment (sindresorhus) [inline index.js]: Why `variant`?

**Comment by sindresorhus** [comment]
> Can we remove them?

👍

File: index.js (hunk 1 above)
Comment (nuno-morais) [reply index.js]: Following the documentation, the ```variant``` has the largest range of differences between strings. It's similar to the ```String.prototype.localeCompare```. 
Do you disagree?

Comment (sindresorhus) [reply index.js]: Good point, but I think we should instead set the `usage` option to `sort`, which also makes it default to `variant`.

Comment (nuno-morais) [reply index.js]: It's not necessary. The default values are ```sort``` and as you said, ```variant```. I deleted the options.

**Comment by sindresorhus** [comment]
> tests are failing on node v0.10 and v0.12 because of xo. Can we remove them?

By this, @satazor meant [these](https://github.com/sindresorhus/alpha-sort/blob/6e16f33e92092f3945dc3ac0820b75463b4e713f/.travis.yml#L6-L7) entries in .travis.yml, not XO itself.

**Comment by satazor** [comment]
Cool, this looks good to be merged!

**Comment by satazor** [comment]
@sindresorhus maybe we should add node 7 to Travis too?

**Comment by sindresorhus** [comment]
Nice work @xnramx :)

**Comment by sindresorhus** [comment]
> maybe we should add node 7 to Travis too?

Nah, I usually only add LTS versions to smaller modules. Too much churn otherwise.

**Pull request merged**